File: problems/0_9/3dprinter/3dprinter.cpp
/*
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
*/

#include <bits/stdc++.h>

using namespace std;

int main() {
    int n, min_days, printers, days;

    cin >> n;

    min_days = n;

    for (int days_printer=2; days_printer<n; days_printer++) {
        printers = pow(2, days_printer - 1);
        days = ceil(n * 1.0 / printers * 1.0) + (days_printer - 1);

        if (days <= min_days) {
            min_days = days;
        }
        else {
            break;
        }
    }
    cout << min_days << endl;
}

File: problems/0_9/3dprinter/3dprinter.py
#! /usr/bin/python3
"""
Author: Rudi César Comiotto Modena
Email: rudi.modena@gmail.com
"""
from math import ceil

def main():
    n = int(input())

    min_days = n

    for days_printer in range(2, n):
        printers = 2 ** (days_printer - 1)
        days = ceil(n / printers) + (days_printer - 1)
        if days <= min_days:
            min_days = days
        else:
            break;

    print(min_days)


if __name__ == '__main__':
    main()